### ISSUE TITLE
slugify dependency is not allowing to install kaggle

### ISSUE THREAD

**Issue by kalessin** [Kaggle/kaggle-api#299]
slugify is an old library that doesn't support python3, so I don't get why kaggle, which is a python3 package, is requiring it. When trying to install via pypi, I get this error:

```
Collecting slugify
  Using cached slugify-0.0.1.tar.gz (1.2 kB)
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-y15tsfxk/slugify/setup.py'"'"'; __file__='"'"'/tmp/pip-install-y15tsfxk/slugify/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-69zmzca8
         cwd: /tmp/pip-install-y15tsfxk/slugify/
    Complete output (11 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-y15tsfxk/slugify/setup.py", line 26, in <module>
        version          = get_version(),
      File "/tmp/pip-install-y15tsfxk/slugify/setup.py", line 20, in get_version
        data = read_from(rel_file('src', 'slugify.py'))
      File "/tmp/pip-install-y15tsfxk/slugify/setup.py", line 15, in read_from
        return fp.read()
      File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 254: ordinal not in range(128)
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by quhb2455** [comment]
do yolo solve it?
i am facing same issue

**Comment by kalessin** [comment]
I didn't continue to investigate how to solve, as I opted to use kaggle web page for submitting.

**Comment by quhb2455** [comment]
@kalessin 
Hi! Finally, I figured out this problem.
when i installed Tensorflow object detection api, this error came out.
It was exactly same error message with your error

python3 doesn't support slugiy-0.0.1 , so i searched slugify-0.0.1.tar.gz file on google and installed it by myself
after then Tensorflow object detection api work!

i hope that this way will help you

**Comment by linzino7** [comment]
> I didn't continue to investigate how to solve, as I opted to use kaggle web page for submitting.
@quhb2455 @kalessin  I had same error with you.

The followings are my solution: 
1. Clone kaggle-api master branch to your PC.

        git clone https://github.com/Kaggle/kaggle-api.git

2. Mark  slugify  dependency package from setup.py in  kaggle-api.git folder.

        'six >= 1.10',^M
        'certifi',^M
        'python-dateutil',^M
        'requests',^M
        'tqdm',^M
        'python-slugify',^M
        #'slugify',^M
        'urllib3',^M


3. Install  slugify  by pip3. This is python3 version.

       pip3 install python-slugify

4. install kaggle-api in folder.

       pip3 install -e .


Done.


My environment:
- Ubuntu 18.04
- Python 3.6.9

**Comment by quhb2455** [comment]
@linzino7 
Hi, Thank you for telling me the solution.
When i faced that problem, i was installing tensorflow api.
And i have tried what you did, but that python-slugify package doesnt work for me. Tf only recognized "slugify"

**Comment by dickmao** [comment]
For me the problem manifested on a `kaggle k push`:
```
Traceback (most recent call last):
  File "/home/dick/.local/bin/kaggle", line 8, in <module>
    sys.exit(main())
  File "/home/dick/.local/lib/python3.6/site-packages/kaggle/cli.py", line 67, in main
    out = args.func(**command_args)
  File "/home/dick/.local/lib/python3.6/site-packages/kaggle/api/kaggle_api_extended.py", line 1907, in kernels_push_cli
    result = self.kernels_push(folder)
  File "/home/dick/.local/lib/python3.6/site-packages/kaggle/api/kaggle_api_extended.py", line 1826, in kernels_push
    as_slug = slugify(title)
  File "/home/dick/.local/lib/python3.6/site-packages/slugify.py", line 24, in slugify
    unicode(
NameError: name 'unicode' is not defined
```

**Comment by tobigue** [comment]
I was able to circumvent the `UnicodeDecodeError` and install the package after running `apt-get install -y locales-all` in my Ubuntu Docker container.

**Comment by jplotts** [comment]
Hi everyone - thanks for reporting this issue!  This should be fixed in the 1.5.10 release, which is available now via pip install.  Please let us know if you're still having issues.
